src/homework/sandro/FourInARow/FourInARow.java

```diff
@@ -0,0 +1,305 @@
+                        || field[i + 2][j] == currentplayer
+                        || field[i][j + 1] == currentplayer
+                        || field[i][j + 2] == currentplayer
+                        || field[i][j + 3] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (count == 4) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean detectWinDiagonalback(int[][] field, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < field.length; i++) {
+            if (field[i][field.length - i - 1] == currentplayer
+                    || field[i + 1][field.length - i - 1] == currentplayer
+                    || field[i + 2][field.length - i - 1] == currentplayer
+                    || field[i][field.length - (i + 1) - 1] == currentplayer
+                    || field[i][field.length - (i + 2) - 1] == currentplayer
+                    || field[i][field.length - (i + 3) - 1] == currentplayer) {
+                count++;
+            } else {
+                count = 0;
+            }
+            if (count == 4) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static boolean detectWinRow(int[][] field, int row, int currentplayer) {         //do i need a int row (from "i") or could i just use field.length ?
+        int count = 0;
+        for (int i = 0; i < field[row].length; i++) {
+            if (field[row][i] == currentplayer) {
+                count++;
+            } else {
+                count = 0;
+            }
+            if (count >= 4) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static boolean detectWinCol(int[][] field, int col, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < field[col].length - 1; i++) {  //why field[col].length has to be -1???
+            if (field[i][col] == currentplayer) {           // condition row, did not need -1 ... what differents
+                count++;                                       // is it -1, couse "i" value for col is one to big and exact for row
+            } else {
+                count = 0;
+            }
+            if (count >= 4) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean hasAnyMoreMoves(int[][] field, int row, int col) {
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < col; j++) {
+                if (field[i][j] == 0) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static int readInt(String message) {
+        int result = -1;
+        Scanner sc = new Scanner(System.in);
+        while (result < 0 || result > 6) {
```

Comment:
you use the same logic in line 268, that is always a sign that you should extract it into its own method